compio-driver/src/sys_slice.rs

```diff
@@ -0,0 +1,117 @@
+use compio_buf::{IoBuffer, IoBufferMut, IoVectoredBuf, IoVectoredBufMut};
+
+#[cfg(unix)]
+mod sys {
+    use std::mem::MaybeUninit;
+
+    #[repr(transparent)]
+    pub struct Inner(libc::iovec);
+
+    impl Inner {
+        pub fn new(ptr: *mut MaybeUninit<u8>, len: usize) -> Self {
+            Self(libc::iovec {
+                iov_base: ptr as *mut libc::c_void,
+                iov_len: len,
+            })
+        }
+    }
+}
+
+#[cfg(windows)]
+mod sys {
+    use std::mem::MaybeUninit;
+
+    // Copied from std
+    #[repr(C)]
+    #[allow(clippy::upper_case_acronyms)]
+    struct WSABUF {
```

Comment:
If it's in `compio-driver`, the real WSABUF should be used.

https://docs.rs/windows-sys/latest/windows_sys/Win32/Networking/WinSock/struct.WSABUF.html